src/Microsoft.IdentityModel.Tokens/LogMessages.cs

```diff
@@ -123,6 +123,7 @@ internal static class LogMessages
         public const string IDX10615 = "IDX10615: Encryption failed. No support for: Algorithm: '{0}', SecurityKey: '{1}'.";
         public const string IDX10616 = "IDX10616: Encryption failed. EncryptionProvider failed for: Algorithm: '{0}', SecurityKey: '{1}'. See inner exception.";
         public const string IDX10617 = "IDX10617: Encryption failed. Keywrap is only supported for: '{0}', '{1}' and '{2}'. The content encryption specified is: '{3}'.";
+        public const string IDX10619 = "IDX10619: Decryption failed. Enc Algorithm: '{0}' is not supported by the given CryptoProviderFactory";
```

Comment:
There are two encryption parameters in play with JWE.
One algorithm 'alg' describes how to think about the ContentEncryptionKey. Is it wrapped, direct key?
The other algorithm 'enc' describes the encryption algorithm of the content, the json payload.
If we included the 'alg' and 'enc' in the error message this would give the user more information to understand what went wrong.
We log if we can't obtain the ContentEncryptionKey due to keywrap algorithms, we may not bubble that into the error message displayed in the exception.

As a side issue, there seems to be duplicate code relating to JWE in JwtSecurityTokenHandler and JsonWebTokenHandler.

Reply:
Great points.

This would be a good opportunity to cut down on the duplication

Reply:
@brentschmaltz although I agree the duplication isn't great, it would be a fairly large change. I propose we de-duplicate as a targeted, specific, and separate PR.